File: solutions/cpp/rna-transcription/1/rna_transcription.cpp
#include "rna_transcription.h"

namespace rna_transcription {
    char to_rna(char dna) {
        if (dna=='G') return 'C';
        if (dna=='C') return 'G';
        if (dna=='T') return 'A';
        return 'U';
    }
    std::string to_rna(std::string dna) {
        std::string rna="";
        for (size_t i=0;i<dna.length(); i++) {
            if (dna.substr(i,1)=="G") rna+="C";
            else if (dna.substr(i,1)=="C") rna+="G";
            else if (dna.substr(i,1)=="T") rna+="A";
            else rna+="U";
        }
        return rna;
    }
}  // namespace rna_transcription

File: solutions/julia/rna-transcription/1/rna-transcription.jl
function to_rna(dna::String)
    out::String=""
    for c::Char in dna
        if c=='G'
             out *= "C"
        elseif c=='C'
            out *= "G"
        elseif c=='T'
            out *= "A"
        elseif c=='A'
            out *= "U"
        else
            throw(ErrorException(dna))
        end
    end
    return out
end

